src/main/java/io/github/dgroup/enumerable4j/Enumerable.java

```diff
@@ -376,4 +379,32 @@ default boolean one(Predicate<X> first, Predicate<X>... other) {
         }
         return out;
     }
+
+    /**
+     * Returns an enumerable of collections which contains each element of original collection
+     *  and corresponding elements from each argument collections.
+     * The length of the resulting enumerable is {@link Enumerable#size}.
+     * If the size of any argument is less than {@link Enumerable#size}, null values are supplied.
+     * @param first The array of enumerable to merge.
+     * @param other The array of enumerable to merge.
+     * @return The enumerable.
+     */
+    default Enumerable<? extends Enumerable<X>> zip(Enumerable<X> first, Enumerable<X>... other) {
+        final List<Enumerable<X>> enms = new ArrayList<>(0);
+        enms.add(first);
+        enms.addAll(Arrays.asList(other));
+        final Enumerable<Enumerable<X>> out = new Linked<>();
+        for (int idx = 0; idx < this.size(); ++idx) {
+            final Linked<X> inner = new Linked<>(((Linked<X>) this).get(idx));
+            for (final Enumerable<X> enm : enms) {
+                X val = null;
+                if (enm != null && idx < enm.size()) {
+                    val = ((Linked<X>) enm).get(idx);
```

Comment:
Use iterator#next instead of type casting and access by index

Reply:
we need  access by index here - it's the same number of elements in the origin 'this' collection and priveded collections.

Reply:
I removed type casting, but left access by index: we need it to access elements with the same index in the different provided collections

Reply:
so, I used `org.cactoos.list.ListOf#ListOf` because `iterator#next` returns `Collection`